server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -271,7 +272,7 @@ public static StartRecoveryRequest getStartRecoveryRequest(Logger logger, Discov
     }
 
     public interface RecoveryListener {
-        void onRecoveryDone(RecoveryState state);
+        void onRecoveryDone(RecoveryState state, ShardLongFieldRange timestampMillisFieldRange);
```

Comment:
maybe `ShardLongFieldRange` belongs to `RecoveryState`?

Reply:
Hmm, `RecoveryState` is really just about progress tracking, I'd prefer to keep this separate. If we were passing it around everywhere then I might change my thinking, but most places that want a `RecoveryState` don't care about the timestamp range.